### ISSUE TITLE
Compilation Error with GCC 8

### ISSUE THREAD

**Issue by panthomakos** [zk-ruby/zookeeper#85]
Gem compiles fine with GCC 7 but fails with GCC 8 with this error. It is reproducible on Ruby 2.4 as well as 2.3:

```
make  2>&1
make  all-am
make[1]: Entering directory '$HOME/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/zookeeper-1.4.11/ext/zkc-3.4.5/c'
/bin/sh ./libtool --tag=CC --mode=compile gcc -DHAVE_CONFIG_H -I. -I. -I.  -I./include -I./tests -I./generated  -Wall -Werror  -g -O2 -D_GNU_SOURCE -c -o zookeeper.lo `test -f 'src/zookeeper.c' || echo './'`src/zookeeper.c
mkdir .libs
 gcc -DHAVE_CONFIG_H -I. -I. -I. -I./include -I./tests -I./generated -Wall -Werror -g -O2 -D_GNU_SOURCE -c src/zookeeper.c  -fPIC -DPIC -o .libs/zookeeper.o
src/zookeeper.c: In function 'format_endpoint_info':
src/zookeeper.c:3440:21: error: '%d' directive writing between 1 and 5 bytes into a region of size between 0 and 127 [-Werror=format-overflow=]
     sprintf(buf,"%s:%d",addrstr,ntohs(port));
                     ^~
src/zookeeper.c:3440:17: note: directive argument in the range [0, 65535]
     sprintf(buf,"%s:%d",addrstr,ntohs(port));
                 ^~~~~~~
src/zookeeper.c:3440:5: note: 'sprintf' output between 3 and 134 bytes into a destination of size 128
     sprintf(buf,"%s:%d",addrstr,ntohs(port));
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[1]: *** [Makefile:754: zookeeper.lo] Error 1
make[1]: Leaving directory '$HOME/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/zookeeper-1.4.11/ext/zkc-3.4.5/c'
make: *** [Makefile:468: all] Error 2
*** extconf.rb failed ***
```

**Comment by deric** [comment]
I'm having the same issue on `ruby 2.5.3p105`

**Comment by johnnywiller** [comment]
Having the same error

Ruby 2.5.3p105 (2018-10-18 revision 65156) [x86_64-linux]
Fedora 29
Kernel version 4.19.13-300.fc29.x86_64

**Comment by kostix** [comment]
On a Debian Buster system this can be sorted out by doing
```
$ sudo apt install gcc-7
$ CC=/usr/bin/gcc-7 bundle install zookeeper --version 1.4.11
```

`gcc-6` will also work.

**Comment by drrzmr** [comment]
To suppress the gcc error:
```
$ CFLAGS=-Wno-error=format-overflow gem install zookeeper --version 1.4.11
```

**Comment by massa1240** [comment]
Same using:
gcc version 9.1.0 (GCC) 
5.1.14-arch1-1-ARCH
ruby 2.3.3p222 (2016-11-21 revision 56859) [x86_64-linux]

supressing the gcc error worked for me :+1:

**Comment by wibotwi** [comment]
Worked also for kafkat:
CFLAGS=-Wno-error=format-overflow  gem install kafkat

**Comment by okupriyanov** [comment]
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=897892
and patch
https://salsa.debian.org/java-team/zookeeper/commit/8f12c964bbc1fc1c025d5fc5e7011a5a67d75181

**Comment by kostix** [comment]
The `CFLAGS=-Wno-error=...` trick does not work for 1.4.1 and 1.4.3 but works for 1.4.11.

Did not bother to check the versions in between, if any.

Thanks, @drrzmr!

**Comment by land-pack** [comment]
I was trying to modify the buffer size.

```
static const char* format_endpoint_info(const struct sockaddr_storage* ep)
{
    static char buf[256];  // new one
    //static char buf[128]; // orignal
    char addrstr[128];
    void *inaddr;
```

**Comment by land-pack** [comment]
> CFLAGS=-Wno-error=format-overflow gem install zookeeper --version 1.4.11

You should switch to root user, and then do this command.

**Comment by kostix** [comment]
@land-pack, please do not misinform: the `gem` tool is perfectly able to install packages for regular users, so there is absolutely no need to switch to `root` unless the user very well understands what they are after.

**Comment by nickmarden** [comment]
Bumping this issue so that interested parties (and the package maintainers) can take a look at my fix for this issue: https://github.com/zk-ruby/zookeeper/pull/90

**Comment by malmckay** [comment]
We've merged and released @nickmarden 's fix in version [1.5.0](https://github.com/zk-ruby/zookeeper/releases/tag/release%2F1.5.0)
